llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -246,11 +246,27 @@ std::string InstrProfError::message() const {
 
 char InstrProfError::ID = 0;
 
-std::string getPGOFuncName(StringRef RawFuncName,
-                           GlobalValue::LinkageTypes Linkage,
+std::string getPGOFuncName(StringRef Name, GlobalValue::LinkageTypes Linkage,
```

Comment:
If we're going to keep the original name, probably want to at least leave big comments in the header not to use it as it is legacy.

Reply:
Also change the function name to getLegacyPGOFuncName

Reply:
I slightly prefer to use `getPGOFuncName` for `;` format and use `getLegacyPGOFuncName` for `:` format, and chose not to rename (want an NFC patch if rename is the way to go, since there are many existing callers).

I updated header and cpp, and did a minimal rename (this function has 3 callers, one in clang directory and another two in InstrProf.cpp)

Reply:
Is this resolved? It looks like the most recent version of the patch changes getPGOFuncName to getLegacyPGOFuncName for the old format, and uses getIRPGOFuncName for the new format.

Reply:
>  It looks like the most recent version of the patch changes getPGOFuncName to getLegacyPGOFuncName for the old format, and uses getIRPGOFuncName for the new format.

This is intentional.  In this PR, renaming the `getPGOFuncName` (that takes [stringified names](https://github.com/llvm/llvm-project/blob/bc4e0c048aa3cd940b0cea787014c7e8680e5040/llvm/lib/ProfileData/InstrProf.cpp#L249) as argument) to  `getLegacyPGOFuncName` is probably fine since it has three callers.

Prior to this PR both `getPGOFuncName` (that [takes Functions as argument](https://github.com/llvm/llvm-project/blob/bc4e0c048aa3cd940b0cea787014c7e8680e5040/llvm/lib/ProfileData/InstrProf.cpp#L359)) and `getIRPGOFuncName` are called in quite a few places, so changing both names would cause too many diffs in this functional change. To use (simpler) `getPGOFuncName` for new `;` format, I'm thinking a separate NFC patch is the way to go.

Reply:
Oh I missed the fact that one getPGOFuncName interface was left. I am only seeing 2 invocations of that interface outside of the unittest test. I would be in favor of doing renames of both getPGOFuncName interfaces in one patch. A separate NFC patch is a fine option, and keeps this patch just about fixing the ICP breakage caused by the delimiter change.

Reply:
> Is that still the case?

Clang FE-PGO calls this function, but afaik it's used for coverage rather than performance nowadays. I guess if Clang FE-PGO continues to use this function it shouldn't be called `legacy` in its name.

I'm wondering if it makes more sense if I rename `getPGOFuncName` (taking stringified names) to `getClangPGOFuncName` in this PR? Thoughts?

Reply:
undo the rename.

I didn't find a way to configure the Github notifications for real-time comments. Guess I'll need to combine email notifications.